src/rgw/rgw_sal_motr.cc

```diff
@@ -57,6 +61,168 @@ static std::string motr_global_indices[] = {
   RGW_MOTR_BUCKET_HD_IDX_NAME
 };
 
+struct MotrFidComparator {
+  bool operator()(const struct m0_fid& left, const struct m0_fid& right) const {
+    return left.f_container < right.f_container ||
+	   (left.f_container == right.f_container && left.f_key < right.f_key);
+  }
+};
+static map<struct m0_fid, MotrWatcher*, MotrFidComparator> motr_watchers;
+
+// process_fdmi_record() can't be implemented as MotrWatcher's member function
```

Comment:
Suggestion: "cannot be implemented as a method of MotrWatcher class, because..."